bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -682,10 +683,13 @@ private void onResource(Resource resource, boolean updateChannels) {
                 updateDependenciesTask = scheduler.submit(() -> updateDependencies());
             }
         } else {
+            if (SUPPORTED_SCENE_TYPES.contains(resource.getType())) {
+                resourceConsumed = checkSceneResourceAddDelete(resource);
+            }
             Resource cachedResource = getResourceFromCache(resource);
             if (cachedResource != null) {
                 Setters.setResource(resource, cachedResource);
-                resourceConsumed = updateChannels && updateChannels(resource);
+                resourceConsumed = (updateChannels ? updateChannels(resource) : true) || resourceConsumed;
```

Comment:
Now reading this logic again, I'm wondering if this scenario is correct:
- A scene was added or deleted, so `resourceConsumed` was already set to `true` in line 687.
- `updateChannels` is `true`.
- `updateChannels(resource)` returns `false`.
- **Result:** `resourceConsumed` is `true`.

Is this expected?

Reply:
^
The bridge handler passes every resource to all thing handlers in turn. And in a normally configured system, at most one thing will “consume” that resource. The purpose of the log message is to allow us to keep track of which thing handler resource did consume the given resource. And under the term “consume” I would include a) added it to cache, b) deleted it from cache, c) updated the cache, or d) updated the channels from cache. Your edge case is a kind of “Schroedinger” case where the resource was both “consumed” (under c) and not “consumed” (under d). And in that case I think it would be proper to NOT log that it was “consumed”.

Reply:
Thanks for reminding me that it actually only makes a difference for debug logging, so really minor. 😄 Nevertheless, just to be sure:
> And in that case I think it would be proper to NOT log that it was “consumed”.

This doesn't seem to be what the code is doing now. If I read it correctly, it WILL log that it was consumed when **c** and not **d**. Do you want to change this or keep it as is?

Reply:
@jlaur I just made a commit that tries to resolve the logging of Schroedingers cat cases

Reply:
Very comprehensive!